### ISSUE TITLE
 Raven and Spotify: A tough couple

### ISSUE THREAD

**Issue by feskyde** [solus-project/budgie-desktop#325]
As you can see on [this video](https://youtu.be/fViuqANSsoA), Raven's MPRIS2 applet and Spotify can't work well (or as with Rhythmbox)

- Raven can not show the song data (it shows nothing)
- Previous/Next tracks buttons are unavailable
- Notifications are shown as a "generic application" thing (and also not show the media player buttons)

**Comment by ikeydoherty** [comment]
Please let me know if you're using a git build or 10.2.2 - because I'm 99% certain this is fixed in git

https://github.com/solus-project/budgie-desktop/compare/v10.2.2...master

**Comment by mank319** [comment]
I have installed 'budgie-desktop' via the AUR which uses the 10.2.2 release and I can confirm that song data is shown correctly (except album art). the *Play/Pause* button works however skipping songs or going back is not possible.

**Comment by mank319** [comment]
**Update:** I just recognized that *Next/Previous* controls are available and functional now for some reason!
So the only thing missing is album cover art :+1: 
I am still on 10.2.2 and have no explanation for why it works now. :question: 

I will keep an eye on it and try to reproduce the problem if it appears again.

**Comment by Justinzobel** [comment]
@mank319 please upgrade to 10.2.3 (only released 2 days ago and see if the issue is all cleared up.

**Comment by theSoenke** [comment]
I'm using 10.2.3 and it does not work. I can only pause and play when i click the play button. But it always looks like this when using spotify:

![raven](https://cloud.githubusercontent.com/assets/1757301/13027255/86b22da2-d247-11e5-8bd0-3d1651cd7d4a.png)

**Comment by ikeydoherty** [comment]
MPRIS is indeed the issue:

```
(budgie-panel:28477): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(budgie-panel:28477): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(budgie-panel:28477): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(budgie-panel:28477): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

** (budgie-panel:28477): CRITICAL **: budgie_header_widget_construct: assertion 'text != NULL' failed

(budgie-panel:28477): GLib-GObject-CRITICAL **: g_object_ref_sink: assertion 'G_IS_OBJECT (object)' failed

(budgie-panel:28477): GLib-GObject-WARNING **: invalid (NULL) pointer instance

(budgie-panel:28477): GLib-GObject-CRITICAL **: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(budgie-panel:28477): Gtk-CRITICAL **: gtk_box_pack: assertion 'GTK_IS_WIDGET (child)' failed

(budgie-panel:28477): GLib-GObject-CRITICAL **: g_object_bind_property_full: assertion 'G_IS_OBJECT (source)' failed

** (budgie-panel:28477): CRITICAL **: budgie_header_widget_set_can_close: assertion 'self != NULL' failed

** (budgie-panel:28477): CRITICAL **: budgie_header_widget_get_can_close: assertion 'self != NULL' failed

** (budgie-panel:28477): CRITICAL **: budgie_header_widget_set_text: assertion 'self != NULL' failed

** (budgie-panel:28477): CRITICAL **: budgie_header_widget_set_icon_name: assertion 'self != NULL' failed
** Message: manager.vala:898: Replaced existing budgie-panel
```

**Comment by ikeydoherty** [comment]
We'll do this in the new `mpris-fixes` branch

**Comment by ikeydoherty** [comment]
Basically it turns out that Spotify is a huge douche when it comes to MPRIS:

```
(spotify:6455): GLib-GIO-WARNING **: Value returned from property 'Get' call for 'PlaybackStatus' should be 's' but is 'a{sv}'
```

**Comment by ikeydoherty** [comment]
Sound indicator/wingpanel: https://bugs.launchpad.net/wingpanel-indicator-sound/+bug/1560218
KDE: https://bugs.kde.org/show_bug.cgi?id=351852

TL;DR The MPRIS implementation in Spotify is trashy. But then our MPRIS support is also trashy. So I'll make ours better, and make it workaround Spotify's .. tendencies.

**Comment by ikeydoherty** [comment]
```
(test:7572): GLib-GIO-WARNING **: Value for PropertiesChanged signal with type '(sa{sv}an)' does not match '(sa{sv}as)'
```

Spotify is on major drugs.

**Comment by ikeydoherty** [comment]
For now, Spotify is blacklisted until they get the client fixed
